### PR TITLE
uninstall for installer syncer

### PR DESCRIPTION
**Pull request by ilackarms**
also should prevent reinstalling the same mesh over and over
untested

### CODE REVIEW

File: pkg/install/install_syncer.go [hunk 1]
```diff
@@ -243,3 +264,13 @@ func getMeshObject(install *v1.Install) (*v1.Mesh, error) {
 	}
 	return mesh, err
 }
+
+func uninstallHelmRelease(releaseName string) error {
+	helmClient, err := helm.GetHelmClient()
+	if err != nil {
+		return err
+	}
```
Comment (rickducott) [inline pkg/install/install_syncer.go]: This doesn't clean up any additional resources, such as CRDs, CRBs

File: api/v1/install.proto [hunk 2]
```diff
@@ -34,6 +34,10 @@ message Install {
     HelmChartLocator chartLocator = 6;
 
     Encryption encryption = 7;
+
+    // whether or not this install should be enabled
+    // if disabled, corresponding resources will be uninstalled
+    bool enabled = 12;
```
Comment (rickducott) [inline api/v1/install.proto]: this should default to true

Comment (ilackarms) [reply api/v1/install.proto]: there's a proto tag for that

File: pkg/install/install_syncer.go (hunk 1 above)
Comment (ilackarms) [reply pkg/install/install_syncer.go]: i left this to a post/pre install method on the interface

File: api/v1/install.proto (hunk 2 above)
Comment (ilackarms) [reply api/v1/install.proto]: fixed

**Comment by ilackarms** [comment]
added an assertion in the e2e that  tests uninstall

File: pkg/install/consul/consul_installer.go [hunk 3]
```diff
@@ -45,6 +45,14 @@ func getOverrides(encryption *v1.Encryption) string {
 	return strings.Replace(overridesYaml, "@@MTLS_ENABLED@@", strBool, -1)
 }
 
+func (c *ConsulInstaller) DoPreHelmUninstall() error {
+	return nil
+}
+
+func (c *ConsulInstaller) DoPostHelmUninstall() error {
+	return nil
```
Comment (rickducott) [inline pkg/install/consul/consul_installer.go]: Needs to remove CRB

**Pull request merged**